pilot/pkg/model/push_context.go

```diff
@@ -970,6 +986,32 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthPolicyForProxy returns the matching auth policy for a given proxy
+func (ps *PushContext) AuthPolicyForProxy(service *Service, port int, proxy *Proxy) *authn.Policy {
+	policy := ps.AuthPolicies[service.Hostname][port]
+	authPolicy := policy.Spec.(*authn.Policy)
+	var workloadPolicy *Config
+
+	if len(authPolicy.Targets) > 0 {
+		// TODO(gihanson) how to handle case of multiple workloads per node
+		for _, l := range proxy.WorkloadLabels {
```

Comment:
This seems wrong. If the policy is ps.AuthPolicies[service.Hostname][port], it's already matched by service host + port. Match by label is smt different. Also, for some background, we planned to replace (host) name with labels, but it's not full supported.